include/swupd.h

```diff
@@ -84,6 +84,7 @@ struct file {
 	int last_change;
 	struct update_stat stat;
 
+	unsigned int fd_valid : 1;
```

Comment:
@pohly I am open to adding file initializer helper functions, for both a struct file on the stack, and dynamically allocated.